mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb

```diff
@@ -1544,6 +1543,24 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 	// Enable partial mode for the resource since it is possible
 	d.Partial(true)
 
+	if d.HasChange("name") {
+		nameV1 := d.Get("name").(string)
+		req := compute.InstancesSetNameRequest{Name: nameV1}
+
+		op, err := config.NewComputeClient(userAgent).Instances.SetName(
+			project, zone, nameV1, &req).Do()
+		if err != nil {
+			return fmt.Errorf("Error updating instance name: %s", err)
+		}
+
+		d.SetId(fmt.Sprintf("projects/%s/zones/%s/instances/%s", project, zone, nameV1))
```

Comment:
We'll want to update the id after the operation finishes, as we may fail during the LRO. There's no simple way to entirely stop losing track of resources depending on where failures occur- that's part of the difficulty of changing identity fields.

Reply:
makes sense. Putting it as the last statement